suite/chaos/reorg_check.go

```diff
@@ -0,0 +1,205 @@
+package chaos
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/chaos/experiments"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/config"
+	"github.com/smartcontractkit/integrations-framework/environment"
+	"golang.org/x/sync/errgroup"
+	"math/big"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	MinerIDTemplate    = "miner-%d"
+	ReorgCheckAttempts = 15
+	ReorgCheckInterval = 10 * time.Second
+)
+
+// ReorgChecker collects data to verify reorged blocks
+type ReorgChecker struct {
+	// main client that is connected to the Chainlink node on which we perform reorg
+	Client client.BlockchainClient
+	Env    environment.Environment
+	// Other miners clients
+	AltClients []client.BlockchainClient
+	FromBlock  uint64
+	HeadersMu  *sync.Mutex
+	Headers    map[uint64]map[string]*types.Header
+	TXHashes   []common.Hash
+}
+
+// NewReorgChecker created new re-org checker connected to all miner nodes
+func NewReorgChecker(connectedClient client.BlockchainClient, env environment.Environment, cfg *config.Config, altMinersPort uint16) (*ReorgChecker, error) {
+	sd, err := env.GetAllServiceDetails(altMinersPort)
```

Comment:
This ties into multi-network support ultimately, so rather than getting all of the service details in here and creating new clients, we should alter the overall setup of the framework to handle multiple clients.

For example, in `environment.go` we should change `NewBlockchainClient` to `NewBlockchainClients` to which it inits all the available Geth clients in a cluster, meaning you don't have to do it here and just use env directly.